src/Microsoft.AspNet.Identity/TaskExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Identity
+{
+    public static class TaskExtensions
+    {
+        public static async Task<IdentityResult> WithLoggingAsync<TUser>(this Task<IdentityResult> identityResult, UserManager<TUser> userManager,TUser user,
+            [System.Runtime.CompilerServices.CallerMemberName] string methodName = "") where TUser :class
+        {
+            return await (await identityResult).WithLoggingAsync(userManager, user, methodName);
```

Comment:
Well, await (await ...) just happens to be what you need sometimes although this is the one you could probably improve a bit with `ContinueWith()` (@suhasj note that the task in the delegate argument of `ContinueWith()` is already completed, or failed).

I don't completely hate the `WithLogging()` experiment. It has a good vibe in that it makes Logging a separate "aspect" that only applies to the results of operations. The one thing that makes me think it is isn't the right way to go is that on every single call we are passing the "this" argument, e.g. the UserManager, RoleManager, SignInManager instance. It is a clear smell that an instance method on those (the way we had it before) would make the most sense.